packages/experience-builder-sdk/src/core/preview/designComponentUtils.ts

```diff
@@ -28,6 +28,9 @@ export const resolveDesignComponent = ({
   }
 
   const componentFields = (designComponent as ExperienceEntry).fields as unknown as Composition;
+  if (!componentFields) {
+    return node;
+  }
```

Comment:
We talked about this in the pairing session. IIRC `fields` would only be empty if it was a `Link` which will happen on CMA but not on CPA/ CDA.
The type of `usedComponents` is already correct in that regard: `Array<Link<'Entry'> | ExperienceEntry>`

Instead of casting back and forth, I would rewrite it this way so TS can infer which of both scenario we're in:
```suggestion
  if (!('fields' in designComponent)) {
    return node;
  }
  const componentFields = designComponent.fields;
```